presto-orc/src/main/java/com/facebook/presto/orc/stream/DecimalStream.java

```diff
@@ -48,68 +49,76 @@ public void seekToCheckpoint(DecimalStreamCheckpoint checkpoint)
         input.seekToCheckpoint(checkpoint.getInputStreamCheckpoint());
     }
 
-    // This comes from the Apache Hive ORC code (see org.apache.hadoop.hive.ql.io.orc.SerializationUtils.java)
-    public BigInteger nextBigInteger()
+    public void nextLongDecimal(Slice result)
             throws IOException
     {
-        BigInteger result = BigInteger.ZERO;
-        long work = 0;
-        int offset = 0;
         long b;
+        long offset = 0;
+        long low = 0;
+        long high = 0;
         do {
             b = input.read();
-            if (b == -1) {
-                throw new OrcCorruptionException("Reading BigInteger past EOF from " + input);
-            }
-            work |= (0x7f & b) << (offset % 63);
-            if (offset >= 126 && (offset != 126 || work > 3)) {
+            if (((offset == 126) &&
+                    (((b & 0x80) > 0) || ((b & 0x7f) > 3)))) {
```

Comment:
Remove some brackets for readability `  if (offset == 126 && ((b & 0x80) > 0 || (b & 0x7f) > 3))`